Dockerify

revolunet opened the pull request.

Commit:
fix(docker): restore docker setup

Commit:
fix(docker): restore docker setup

Commit:
chore(yarn): allow experimental workspaces

lionelB commented on README.md:
```diff
@@ -1,38 +1,59 @@
+## Setup
 
-L'intégration HTML/CSS est réalisée et documentée dans un autre dépôt de code : [code-du-travail-css](https://github.com/SocialGouv/code-du-travail-css).
+Editer `.env` et `docker-compose.override.yml` puis lancer `docker-compose up`
```
normalement ca sera `yarn dev`

lionelB replied:
> normalement ca sera yarn dev 
c'est pas si pratique d'avoir les 3serveur qui tourne dans le meme term en fait :)

douglasduteil commented on .yarnrc:
```diff
@@ -0,0 +1 @@
+  workspaces-experimental true 
```
really required ?

douglasduteil replied:
see https://yarnpkg.com/lang/en/docs/workspaces/#toc-tips-tricks

lionelB replied:
it was to avoid warning message.

douglasduteil commented on docker-compose.yml:
```diff
@@ -17,20 +16,18 @@ services:
+    environment:
+      - PYTHONPATH=.
+      - ES_HOST=elasticsearch
+      - PORT=9200
```
ES_PORT ?

douglasduteil commented on packages/code-du-travail-frontend/Dockerfile:
```diff
@@ -1,23 +1,42 @@
 
 COPY package.json /app/package.json
-COPY package-lock.json /app/package-lock.json
+#COPY yarn.lock /app/yarn.lock
```
to remove ?

revolunet replied:
will be back when yarn.lock will be versionned

douglasduteil replied:
we should do it now then...

douglasduteil replied:
\ping @revolunet 
Same goes for `/packages/code-du-travail-api/Dockerfile` ?

douglasduteil reviewed (changes requested):
:whale:

Commit:
fix(docker): fix ES_PORT, move data docker scripts to data

Commit:
chore: ignore docker-compose.yml

lionelB committed:
add launch shortcuts for  subprojects

douglasduteil committed:
chore: remove yarnrc

lionelB committed:
fix fragmentSize param name in suggest endpoint

douglasduteil committed:
style(data): add missing last EOL

lionelB committed:
Merge pull request #163 from SocialGouv/launch-shortcuts

add launch shortcuts for  subprojects

douglasduteil committed:
chore: setup env on post install (#165)

lionelB committed:
fix(dev): update dev in npm run script

remove code-du-travail-data/.env file since its no more used
add code-du-travail-api/.env file with default value for dev env

revolunet commented:
je propose qu'on merge et on améliore ensuite

douglasduteil commented on packages/code-du-travail-api/.gitignore:
```diff
@@ -63,8 +63,5 @@ typings/
 .yarn-integrity
 
-# dotenv environment variables file
-.env
```
Same should be done in `/packages/code-du-travail-css/.gitignore` ?

lionelB replied:
agree, since code-du-travail-css was not using a env file, I forget to update it :)

lionelB committed:
fixup! fix(dev): update dev in npm run script

Commit:
fix(dockerify): add missing .env

lionelB commented on packages/code-du-travail-data/.env:
```diff
@@ -0,0 +1,3 @@
+PYTHONPATH=.
+ES_HOST=elasticsearch
+ES_PORT=9200
```
I'm sorry,  I thought these files was unused anymore since we handle the env globally in `docker-compose.yml`

lionelB replied:
In which case we need it ?

revolunet replied:
indeed, they are overriden by the root `docker-compose.yml`, we shouldnt need this except for explicitness maybe. will remove that!

Commit:
Revert "fix(dockerify): add missing .env"

This reverts commit c612762911d24edc61e63ae5fcd0344292013e88.
